assets/app/scripts/directives/fromFile.js

```diff
@@ -236,9 +236,15 @@ angular.module("openshiftConsole")
           return DataService.get(resourceGroupVersion, item.metadata.name, $scope.context, {errorNotification: false}).then(
             // resource does exist
             function(resource) {
-              if (item.kind !== "Template") {
+              // For Template update it's Objects, Paramaters and Labels fields. For other resources update their Spec.
+              if (item.kind === "Template") {
+                resource.objects = item.objects;
+                resource.parameters = item.parameters;
+                resource.labels = item.labels;
```

Comment:
Right, I understand the behavior, but how is it coded?

I'm just hesitant to begin making lists of fields to update since it will be hard to maintain and get out of sync when the API changes.